src/updater/updater.cpp

```diff
@@ -62,18 +62,34 @@ inline EdgeWeight convertToDuration(double speed_in_kmh, double distance_in_mete
     if (speed_in_kmh <= 0.)
         return MAXIMAL_EDGE_DURATION;
 
-    const double speed_in_ms = speed_in_kmh / 3.6;
-    const double duration = distance_in_meters / speed_in_ms;
-    return std::max<EdgeWeight>(1, static_cast<EdgeWeight>(std::round(duration * 10.)));
+    const auto speed_in_ms = speed_in_kmh / 3.6;
+    const auto duration = distance_in_meters / speed_in_ms;
+    return std::max(1, boost::numeric_cast<EdgeWeight>(std::round(duration * 10.)));
 }
 
-inline EdgeWeight convertToWeight(double weight, double weight_multiplier, EdgeWeight duration)
+inline EdgeWeight convertToWeight(const SpeedSource &value,
+                                  double distance_in_meters,
+                                  const extractor::ProfileProperties &profile_properties)
 {
-    if (std::isfinite(weight))
-        return std::round(weight * weight_multiplier);
+    double rate = value.rate;
+    if (!std::isfinite(rate))
+    {
+        if (!profile_properties.fallback_to_duration)
```

Comment:
Instead of printing this for every line what about adding a counter, and only printing this warning one with a count like "used rate fallback to speed for X entries"? This might spam the logs, given we have CSV files with a lot of entries.